core/ui/src/lib/components/repo/ModifiedFilesCard.svelte

```diff
@@ -496,7 +503,7 @@
 									on:click={() => {
 										handleToggleAllFilesInChangeset(index);
 										changeSets[index] = {
-											...changeSet,
+											...changeSets[index],
```

Comment:
`changeSet` in this scope is actually the changeset as filtered by the searchbar, so we were actually modifying the changeset data based on search results. the would result in the modified files card only showing selected files even when the search bar was empty and there were other modified files.